scripts/get_indexfiles.py

```diff
@@ -78,9 +84,27 @@ def main():
         format='%(asctime)s - %(process)s - %(levelname)s - %(message)s'
     )
 
-    base_url = 'https://data.commoncrawl.org/cc-index/collections/' \
+    base_url = f'https://data.commoncrawl.org/cc-index/collections/' \
                f'{args.collection}/indexes/'
 
+    if args.patterns:
+        patterns = args.patterns
+    else:
+        patterns = []
+        with open(args.pattern_file, 'rt') as pf:
+            for line in pf:
+                patterns.append(line.strip())
+
+    patterns_as_lists = []
+    for pattern in patterns:
+        pattern_list = pattern.split('.')
+        pattern_list.reverse()
+        if pattern_list[-1] == '*':
+            pattern_list.pop()
+        patterns_as_lists.append(pattern_list)
```

Comment:
`main()` is getting longer with all these new functionalities added, so I propose to extract this bit out into a separate function. You can do so with other parts as you see fit.

Reply:
I extracted it out and then used a list comprehension! :)